arch/x86/kvm/vmx/pkvm/hyp/ept.h

```diff
@@ -33,7 +33,7 @@ int pkvm_host_ept_init(struct pkvm_pgtable_cap *cap, void *ept_pool_base,
 		unsigned long ept_pool_pages);
 int handle_host_ept_violation(unsigned long gpa);
 void pkvm_flush_host_ept(void);
-int pkvm_shadow_ept_pool_init(void *ept_pool_base, unsigned long ept_pool_pages);
+int pkvm_shadow_pgt_pool_init(void *ept_pool_base, unsigned long ept_pool_pages);
```

Comment:
Just a note: the name "shadow_pgt" is gonna be a bit ambiguous if we merge my patch https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/4231786 which adds a pool of shadow IOMMU page tables.

Reply:
I see. The reason of changing to shadow_pgt is to make it common to both shadow ept and pgstate pgt as they are using the same pool which is shadow ept pool. With introducing iommu shadow page table pool by your path, shadow_pgt seems also common. In this case, do you think if keep the current name is better for this function?
BTW, I think keep using "shadow_pgt" inside ept.c should have no ambiguous. What do you think?

Reply:
Yes, keeping using "shadow_pgt" for private functions inside ept.c seems perfectly fine. Using it for this public pkvm_shadow_pgt_pool_init() function would be slightly ambiguous with regard to shadow iommu page tables from my patch, but it is rather fine with me too (I have no better suggestions anyway), so please feel free to ignore  this remark.

I'm not sure what exactly do you mean by "With introducing iommu shadow page table pool by your path, shadow_pgt seems also common". Maybe you are suggesting to extend this shadow_pgt pool to use it for allocating shadow iommu page tables as well, together with shadow ept and pgstate pgt, instead of adding a new pool like in my patch? It seems not so important whether we use a separate pool or not, that said, unlike shadow_ept and pgstate_pgt, shadow iommu page tables are not tied with shadow VMs, since they are meant precisely for ptdevs *not* attached to any shadow VM.

Reply:
For the comment "With introducing iommu shadow page table pool by your path, shadow_pgt seems also common", as you mentioned the name of shadow_pgt is a bit ambiguous with your patch, I though you mean shadow_pgt is common to shadow_ept/pgstate_pgt/shadow_iommu_pgt, so pkvm_shadow_pgt_pool_init may make people think this function is used for shadow_iommu_pgt as well.

I can make the change to keep the name of pkvm_shadow_ept_pool_init(), so that there is no ambiguous.

Reply:
I think we shall try to unify these two pgtables, pgstate_pgt is designed as pgstate table and can be reused as IOMMU page table. For legacy IOMMU support, could we use it as host VM's pgstate table instead of host ept?

Reply:
After 2nd think, seemed for untrusted devs, it's not able to use IOMMU page table work as pgstate table - as there could be multiple untrusted devices with multiple IOMMU page tables. 
But still, we should be able to unify IOMMU shadow page table into shadow pgt pool, right?

Reply:
@cxdong,
> I though you mean shadow_pgt is common to shadow_ept/pgstate_pgt/shadow_iommu_pgt, so pkvm_shadow_pgt_pool_init may make people think this function is used for shadow_iommu_pgt as well.

Right, that is what I meant. Agree, keeping the old name pkvm_shadow_ept_pool_init() would avoid this ambiguity (I don't insist on it though).

@JasonChenCJ,
Yes, there may be an arbitrary number of untrusted devices using different vIOMMU page tables, independently of the number of shadow VMs, so unifying it with pgstate_pgt doesn't sound quite feasible.

Regarding which pool to use for IOMMU shadow page table, I see such options:

1. add a new pool (as in my patches)
2. extend the shadow pgt pool (since these pgtables are a kind of shadow pgt)
3. extend the iommu pool (since these pgtables are a part iommu functionality)

and IIUC, it's just a matter of taste which of these 3 options to choose, it would not have an impact on any functionality. Please let me know if you think some of these options is better than others.

Reply:
I prefer to #2.

Reply:
@JasonChenCJ Ok, just to confirm: you'd prefer if I rework shadow iommu page tables in my patches, to use the shadow ept pool instead of adding a new pool, correct?
Now it seem like a good idea to me too: I can reuse the same mm_ops as for pgstate_pgt instead of reimplementing mm_ops for shadow iommu page tables, and also can reuse pkvm_pgstate_pgt_update_coherency().

Reply:
@semihalf-maluka-dmytro , yes, we can reuse what we had now. Thanks!